Enhancements to support inet6/bridge/evpn in debugroute vertex

srinivn opened the pull request:
Following routing tables are placed in the differen modules as follows
In Agent,
- inet-unicast
- inet6-unicast
- inet4-multicast
- bridge-route
- evpn-route

In Vrouter(kernel),
- inet
- inet6
- bridge

In Controller,
- inet
- inet6
- evpn

Based on the avaialbility of those tables the given prefix along with
the routing table type, will serach the correpsonding table.

The sample commands to search different routing tables as follows,
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 02:1c:bf:17:9d:a7 --rt_type bridge
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 10.10.10.3 --rt_type inet
python debugroute.py --ri_fqname default-domain:admin:ipv6_nw:ipv6_nw
--prefix fd11::3 --rt_type inet6
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 02:1c:bf:17:9d:a7,10.10.10.3 --rt_type evpn

srinivn committed:
Enhancements to support inet6/bridge/evpn in debugroute vertex

Following routing tables are placed in the differen modules as follows
In Agent,
- inet-unicast
- inet6-unicast
- inet4-multicast
- bridge-route
- evpn-route

In Vrouter(kernel),
- inet
- inet6
- bridge

In Controller,
- inet
- inet6
- evpn

Based on the avaialbility of those tables the given prefix along with
the routing table type, will serach the correpsonding table.

The sample commands to search different routing tables as follows,
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 02:1c:bf:17:9d:a7 --rt_type bridge
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 10.10.10.3 --rt_type inet
python debugroute.py --ri_fqname default-domain:admin:ipv6_nw:ipv6_nw
--prefix fd11::3 --rt_type inet6
python debugroute.py --ri_fqname default-domain:admin:local:local
--prefix 02:1c:bf:17:9d:a7,10.10.10.3 --rt_type evpn

smurugap commented on introspect.py:
```diff
@@ -237,16 +237,33 @@ def matched_adjacencies(self, node_type, fq_name_str):
+        elif ((rt_type == 'bridge') or (rt_type == 'evpn')):
+            rt_name = 'evpn'
+        else:
+            rt_name = None
```
Believe we should return None if rt_type is not one of inet/inet6/evpn/bridge

srinivn replied:
have added the following changes to handle this case
        else:
            rt_name = None
            routes = None
            return(False, routes)

With out this change also, existing code returns the False and behavior is same. How to do increment commit to accommodate this change? Does it require one more pull request?

smurugap commented on debugroute.py:
```diff
@@ -32,11 +32,12 @@ class debugVertexRoute(baseVertex):
         self.ri_fqname = ri_fqname
         self.vn_fqname = vn_fqname
         self.prefix = prefix
+        self.rt_type = rt_type
```
Can we determine the route type if not provided?
You may refer https://github.com/Juniper/contrail-test-ci/blob/master/tcutils/util.py#L586

srinivn replied:
I have also kept similar logic initially. But evpn has the combination of both the addresses(mac+ip). so have made the explicit routing type

smurugap commented on introspect.py:
```diff
@@ -357,38 +372,79 @@ def get_vrf_id(self, vrf_fqname):
+        elif rt_type == 'evpn':
+            rt_name = 'evpn.route'
+        else:
+            rt_name = None
```
Same here, we should be bailing out in case of wrong rt_type

srinivn replied:
Followed the same as comment 1

srinivn committed:
Addressed the following review comments

Comment:
Believe we should return None if rt_type is not one of
inet/inet6/evpn/bridge

Fix:
have added the following changes to handle this case
else:
rt_name = None
routes = None
return(False, routes)

With out this change also, existing code returns the False and behavior
is same.
